ot/lp/_network_simplex.py

```diff
@@ -272,38 +289,78 @@ def emd(
     ot.optim.cg : General regularized OT
     """
 
-    a, b, M = list_to_array(a, b, M)
-    nx = get_backend(M, a, b)
+    edge_sources = None
+    edge_targets = None
+    edge_costs = None
+    n1, n2 = None, None
+
+    # Get backend from M first, then use it for list_to_array
+    # This ensures empty lists [] are converted to arrays in the correct backend
+    nx_M = get_backend(M)
+    a, b = list_to_array(a, b, nx=nx_M)
```

Comment:
move it back above and update list_to_array so that it deos nothing if sparse matrix? here you broque the possibility to give lists to emd see broken test